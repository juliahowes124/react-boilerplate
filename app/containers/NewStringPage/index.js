/*
 * NewStringPage
 *
 * This is the first thing users see of our App, at the '/' route
 *
 */

import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import styled from 'styled-components';
import { useHistory } from 'react-router-dom';

const StyledButton = styled.button`
  background-color: ${props => props.bg};
  color: white;
  border: 2px solid darkgreen;
`;

export default function NewStringPage() {
  const dispatch = useDispatch();
  const history = useHistory();
  const [formData, setFormData] = useState({ string: '' });

  function handleSubmit(evt) {
    evt.preventDefault();
    dispatch({ type: 'ADD_STRING', string: formData.string });
    setFormData({ string: '' });
    history.push('/');
  }

  function handleChange(evt) {
    const { name, value } = evt.target;
    setFormData(data => ({ ...data, [name]: value }));
  }

  return (
    <form onSubmit={handleSubmit}>
      <input name="string" value={formData.string} onChange={handleChange} />
      <StyledButton bg="red">Add</StyledButton>
    </form>
  );
}
