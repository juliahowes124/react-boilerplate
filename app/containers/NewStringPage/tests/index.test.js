import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';

import NewStringPage from '../index';

describe('<NewStringPage />', () => {
  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale="en">
        <NewStringPage />
      </IntlProvider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
