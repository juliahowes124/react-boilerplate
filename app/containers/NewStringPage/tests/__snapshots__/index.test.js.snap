// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewStringPage /> should render and match the snapshot 1`] = `
<form>
  <h1
    class="styledComponents__StyledTitle-sc-1tnxmlh-0 eTIHrx"
  >
    Add a new string
  </h1>
  <div>
    <input
      aria-label="string"
      id="string"
      name="string"
      placeholder="Enter a new string"
      required=""
      value=""
    />
    <button
      class="styledComponents__StyledButton-sc-1tnxmlh-1 fmoCRp"
    >
      Submit
    </button>
  </div>
</form>
`;
